storage/src/main/java/zipkin2/storage/kafka/KafkaSpanStore.java

```diff
@@ -126,78 +129,120 @@
   }
 
   static class GetServiceNamesCall extends KafkaStreamsStoreCall<List<String>> {
-    ReadOnlyKeyValueStore<String, Set<String>> serviceStore;
+    ReadOnlyKeyValueStore<String, String> serviceStore;
 
-    GetServiceNamesCall(ReadOnlyKeyValueStore<String, Set<String>> serviceStore) {
+    GetServiceNamesCall(ReadOnlyKeyValueStore<String, String> serviceStore) {
       this.serviceStore = serviceStore;
     }
 
-    @Override
-    List<String> query() {
-        try {
-          List<String> keys = new ArrayList<>();
-          serviceStore.all().forEachRemaining(keyValue -> keys.add(keyValue.key));
-          return keys;
-        } catch (Exception e) {
-          LOG.error("Error looking up services", e);
-          return new ArrayList<>();
-        }
+    @Override public List<String> query() {
+      try {
+        List<String> serviceNames = new ArrayList<>();
+        serviceStore.all().forEachRemaining(keyValue -> serviceNames.add(keyValue.value));
+        Collections.sort(serviceNames); // comply with Zipkin API
+        return serviceNames;
+      } catch (Exception e) {
+        LOG.error("Error looking up services", e);
+        return new ArrayList<>();
+      }
     }
 
-    @Override
-    public Call<List<String>> clone() {
+    @Override public Call<List<String>> clone() {
       return new GetServiceNamesCall(serviceStore);
     }
   }
 
   static class GetSpanNamesCall extends KafkaStreamsStoreCall<List<String>> {
-    final ReadOnlyKeyValueStore<String, Set<String>> serviceStore;
+    final ReadOnlyKeyValueStore<String, Set<String>> spanNamesStore;
     final String serviceName;
 
-    GetSpanNamesCall(ReadOnlyKeyValueStore<String, Set<String>> serviceStore, String serviceName) {
-      this.serviceStore = serviceStore;
+    GetSpanNamesCall(ReadOnlyKeyValueStore<String, Set<String>> spanNamesStore,
+        String serviceName) {
+      this.spanNamesStore = spanNamesStore;
       this.serviceName = serviceName;
     }
 
-    @Override
-    List<String> query() {
-        try {
-          if (serviceName == null || serviceName.equals("all")) return new ArrayList<>();
-          Set<String> spanNames = serviceStore.get(serviceName);
-          if (spanNames == null) return new ArrayList<>();
-          return new ArrayList<>(spanNames);
-        } catch (Exception e) {
-          LOG.error("Error looking up for span names for service {}", serviceName, e);
-          return new ArrayList<>();
-        }
+    @Override public List<String> query() {
+      try {
+        if (serviceName == null || serviceName.equals("all")) return new ArrayList<>();
```

Comment:
careful with magic words.. if lens is sending the word "all" that's a bug

Reply:
It used to happen with old UI, but will keep an eye to check if happens with lens.